File: Java/Fibonacci.java
public class Fibonacci {

	public static int fibonacci(int n) {
		if (n <= 0) {
			return 0;
		} else if (n == 1) {
			return 1;
		} else {
			return fibonacci(n-1) + fibonacci(n-2);
		}
	}
	public static void main(String[] args) {
		// test
		System.out.println(fibonacci(4));
		System.out.println(fibonacci(12));
	}
}

File: Go/fibonacci.go
package main

import "fmt"

func main() {

	fmt.Print(fibonacci(10)) // 55

}

// Fibonacci is a simple program to learn about recursion
func fibonacci(n int) int {
	if n < 1 {
		return 0
	} else if n < 3 {
		return 1
	}
	return fibonacci(n-1) + fibonacci(n-2)
}
